numpy/lib/arraysetops.py

```diff
@@ -94,7 +94,8 @@ def ediff1d(ary, to_end=None, to_begin=None):
     # force a 1d array
     ary = np.asanyarray(ary).ravel()
 
-    # enforce propagation of the dtype of input
+    # we have unit tests enforcing
+    # propagation of the dtype of input
```

Comment:
Why this change? I am not a big fan of comments that reference other parts of the codebase that could change in the future. Also, the unit tests only check that enforcing propagation is done, they do not enforce it themselves

Reply:
I think this comment change came in as part of the revert (which I subsequently squashed). I agree, the comment is not great.

Reply:
Adjusted.